---
title: "1Chapter 3: Getting data and trees into R"
output: html_notebook
---

<!-- 3.1 Data and tree object types -->
```{r}
class("Darwin")
```
```{r}
print(pi)
```
```{r}
class(pi)
```
```{r}
example.1 <- "6"
print(example.1)
```
```{r}
example.2 <- as.numeric(example.1)
print(example.2)
```
```{r}
class(example.2)
```
```{r}
example.2 * 7
```

3.1.1 Tree classes
```{r}
library(ape)
phy <- ape::rcoal(5)
print(phy)
```
```{r}
str(phy)
```
<!-- 3.2 Sequence data -->
<!--   The BioConductor community is working with R, but python still more popular.  -->
<!--   Packages: seqinr (useful for handling sequences), ape and phangorn (handing data, reading FASTA and NEXUS) -->

<!-- 3.3 Other character data -->
<!-- Can include data such as discrete traits (wings/no wings), continuous (body mass), geographic traits (lat/long, continent). Typically loaded as csv files from other source or from an R package. rOpenSci supports useful packages. Packages for getting data from GBIF, eBIRD, iNaturalist, NCBI. -->

<!-- 3.4 phylogenies -->
```{r}
# most common way to load trees is ape
phy <- ape::read.tree(file='treefile.phy')
```
```{r}
# get a tree in Newick (also Phylip) format (save it as .phy)
phy <- ape::read.tree(text = '((Strix_aluco:4.2,Asio_otus:4.2):3.1,Athene_noctua:7.3);')
plot(phy)
```
<!-- The output class depends on the input. If there is only 1 tree (class = phylo), plot() works. If there's multiple trees (class = multiphylo), plot() cycles through the trees with enter, and comparitive methods fail. read.tree() has an optional keep.multi function. If TURE, the class is always multiphylo, and get the first tree by getting the first element in the returned object -->
```{r}
phy.multi <- ape::read.tree(file = 'treefile.phy', keep.multi = TRUE)
plot(phy.multi)
phy <- phy.multi[[1]]
plot(phy)
```
<!-- NEXUS formatted files have ape::read.nexus() can pull in the trees, and ape::read.nexus.data() can pull in data from a Nexus file. Some valid Nexus file formats still won't work with ape. A more robust function is phylobase::readNexus(). Uses a diff structure to store trees than ape. -->

<!-- tidytree is still fairly new, but is popular for dealing with trees if you like tidyverse. -->

<!-- 3.4.1 Great scientists steal -->
<!-- TreeBase is a repository for phylogenetic trees that publications can post their trees to. Trees can be downloaded and loaded into R using phylobase (the Nexus format used by TreeBase is hard for ape to load). -->
<!-- Open Tree of Life: seeks to synthesize thousands of trees to create a single tree of life. The rotl package can download this synthetic tree or components of it. For most groups the tree is based on taxonomy, not well resolved. Don't use this yet for publication quality trees, but for exploration, sure. rotl lets you download the cache of hand curated trees from the site. You can download chronograms. -->

<!-- Always cite the trees you re-use. If it's hundreds of trees, you can put an appendix in supplemental for all relevant papers. -->
<!-- Tree quality matters: many comparative methods are based on using branch lengths (rates of character evol, diversification rates). If your starting tree is wrong (even just the branch lengths), later downstream analyses are also likely to be wrong.  -->

<!-- 3.5 Reconciling datasets -->
Species names may change: 
  - Species are split into two species
  - Two species are lumped into one species
  - A higher level group is changed
  - An error is fixed









